package-builder/flox-build.mk

```diff
@@ -620,59 +659,68 @@ define NIX_EXPRESSION_BUILD_template =
 	  }' \
 	  pkgs.$(_pname) > $$@
 
+  # Following a successful eval, carry on with building the drvPath directly.
   $($(_pvarname)_buildJSON): $($(_pvarname)_evalJSON)
 	@# Now that we have the metadata, set the _name.
 	$$(eval _name = $$(shell $(_jq) -r '.name' $$<))
+	$$(eval _drvPath = $$(shell $(_jq) -r '.drvPath' $$<))
+	@# Verify that the drvPath still exists in the store because there
+	@# is a very small but non-zero chance that the package has been
+	@# garbage collected since the eval was performed.
+	$$(if $$(wildcard $$(_drvPath)),,\
+	  $$(error $$(_drvPath) has been garbage collected - please try again))
 	@echo "Building $$(_name) in Nix expression mode"
-	$(_V_) $(_nix) build --json -L --out-link $(_result) \
-	  $$(shell $(_jq) -r '.drvPath' $$<)'^*' > $($(_pvarname)_buildJSON)
+	$(_V_) $(_nix) build --json -L --out-link $($(_pvarname)_result) \
+	  $$(_drvPath)'^*' > $$@
 
-  # Note that the result symlink is created as a side-effect of the build.
-  $(_result): $($(_pvarname)_buildJSON)
+  # Recall that the $(_pvarname)_CHECK_RESULT_LINKS target as declared in the
+  # COMMON_BUILD_VARS_template checks that the result symlinks point to the
+  # expected store paths.
 
-  # Similarly note result symlink target for each output apart from "out".
-  $(foreach _output,$(subst $(comma), ,$(_outputs)), \
-    $(if $(filter-out out,$(_output)), \
-      $(_result)-$(_output): $($(_pvarname)_buildJSON)
-    )
-  )
+  # Perform post-build checks common to all build modes.
+  .INTERMEDIATE: $(_pvarname)_CHECK_BUILD
+  $(_pvarname)_CHECK_BUILD: $(_pvarname)_CHECK_RESULT_LINKS
+	@# do something here?
 
   # Harvest the logfile from the build.
-  $($(_pvarname)_logfile): $(_result) FORCE
-	( $(_V_) $(_nix) log $$</. || echo "No logs available" ) > $($(_pvarname)_logfile)
+  $($(_pvarname)_logfile): $($(_pvarname)_buildJSON) $(_pvarname)_CHECK_BUILD
+	$$(eval _drvPath = $$(shell $(_jq) -r '.[0].drvPath' $$<))
+	$(_V_) ( $(_nix) log $$(_drvPath) || echo "No logs available" ) > $($(_pvarname)_logfile)
 
   # Add the log to the store and create a GCRoot for it.
-  $(_result)-log: $($(_pvarname)_logfile)
-	$(_V_) $(_nix) build -L --out-link $(_result)-log \
+  $($(_pvarname)_result)-log: $($(_pvarname)_logfile)
+	$(_V_) $(_nix) build -L --out-link $($(_pvarname)_result)-log \
 	  $$(shell $(_nix) store add-file $($(_pvarname)_logfile))
```

Comment:
doenst this (and other `$(_pvarname)_XXX` files) create a bunch of temporary files in the workif directory?
personally im not even excited about the result links being pread out in the working dir (rather than a specific subdir) but having lots of control files in my working dir seems annoying and leaky of the implementation.

Reply:
No, all of the `$(_pvarname)_*` files are kept in the `$(PROJECT_DIR)`, which is a hashed subdirectory of `$(TMPDIR)`. The only files that reside in the current directory (or rather the "topdir" where the environment resides) are the `result-*` GCroot symlinks.